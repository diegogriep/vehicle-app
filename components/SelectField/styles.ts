import styled from 'styled-components'

export const Wrapper = styled.div``

export const SelectWrapper = styled.div`
  background-color: #fff;
  border: 0.1rem solid #333;
  border-radius: 0.4rem;
  padding: 0.8rem 1.2rem 0.8rem;

  &:focus-within {
    border-color: rgba(142, 216, 248, 0.5);
  }
`

export const Select = styled.select`
  border-radius: 0.4rem;
  color: #666;
  padding: 0;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
`

export const Label = styled.label`
  color: #000;
  cursor: pointer;
  display: inline-block;
`
