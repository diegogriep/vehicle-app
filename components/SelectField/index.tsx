import { useState } from 'react'
import * as S from './styles'
import { SelectFieldProps } from '@/types/types'

const SelectField = ({
  items,
  label,
  labelFor = '',
  initialValue = '',
  placeholder = '',
  updatedValue = '',
  onInput,
  ...props
}: SelectFieldProps) => {
  const [value, setValue] = useState(initialValue)

  const onChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    const newValue = e.currentTarget.value
    setValue(newValue)

    !!onInput && onInput(newValue)
  }

  return (
    <S.Wrapper>
      {!!label && <S.Label htmlFor={labelFor}>{label}</S.Label>}
      <S.SelectWrapper>
        <S.Select
          onChange={onChange}
          {...props}
          value={typeof updatedValue === 'object' ? '' : value}
          id={labelFor}
        >
          <option value="">{placeholder ? placeholder : 'Select'}</option>
          {items.map((item) => (
            <option key={item.value} value={item.value}>
              {item.label}
            </option>
          ))}
        </S.Select>
      </S.SelectWrapper>
    </S.Wrapper>
  )
}

export default SelectField
