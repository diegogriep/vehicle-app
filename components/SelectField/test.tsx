import { render, screen } from '@testing-library/react'

import SelectField from '.'

const items = [
  {
    label: 'BMW',
    value: 'BMW'
  },
  {
    label: 'VW',
    value: 'VW'
  },
  {
    label: 'Ford',
    value: 'Ford'
  }
]

describe('<SelectField />', () => {
  it('should render the combobox', () => {
    const { container } = render(<SelectField items={items} />)

    expect(screen.getByRole('combobox')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
