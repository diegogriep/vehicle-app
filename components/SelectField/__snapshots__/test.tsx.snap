// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectField /> should render the combobox 1`] = `
.c0 {
  background-color: #fff;
  border: 0.1rem solid #333;
  border-radius: 0.4rem;
  padding: 0.8rem 1.2rem 0.8rem;
}

.c0:focus-within {
  border-color: rgba(142, 216, 248, 0.5);
}

.c1 {
  border-radius: 0.4rem;
  color: #666;
  padding: 0;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

<div
  class=""
>
  <div
    class="c0"
  >
    <select
      class="c1"
      id=""
    >
      <option
        value=""
      >
        Select
      </option>
      <option
        value="BMW"
      >
        BMW
      </option>
      <option
        value="VW"
      >
        VW
      </option>
      <option
        value="Ford"
      >
        Ford
      </option>
    </select>
  </div>
</div>
`;
